docker-build make target doesn't work

**What steps did you take and what happened:**

1. run `make docker-build REGISTRY=<myregistry>`

the following error occurs because the go language version doesn't have the updated package.
```shell
 => ERROR [builder 9/9] RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 GO111MODULE=on go build -a -o manager main.go    1.5s
------
 > [builder 9/9] RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 GO111MODULE=on go build -a -o manager main.go:
#17 1.460 /go/pkg/mod/golang.org/x/net@v0.2.0/http2/transport.go:19:2: package io/fs is not in GOROOT (/usr/local/go/src/io/fs)
------
executor failed running [/bin/sh -c CGO_ENABLED=0 GOOS=linux GOARCH=amd64 GO111MODULE=on go build -a -o manager main.go]: exit code: 1
make: *** [Makefile:145: docker-build] Error 1
```

**What did you expect to happen:**

i expected to see `myregistry/cluster-api-kubemark-controller-amd64:dev` when inspecting my images.


**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]


**Environment:**

- cluster-api version: n/a
- cluster-api-provider-kubemark version: d48a3d3dbeff24b6774f57976aa92c76bae81aea
- Kubernetes version: (use `kubectl version`): n/a
- OS (e.g. from `/etc/os-release`): Ubuntu 22.04 LTS

/kind bug
[One or more /area label. See https://github.com/kubernetes-sigs/cluster-api-provider-kubemark/labels?q=area for the list of labels]

--- a/Dockerfile
+++ b/Dockerfile
@@ -15,7 +15,7 @@
 # limitations under the License.
 
 # Build the manager binary
-FROM golang:1.15 as builder
+FROM golang:1.19 as builder
 
 WORKDIR /workspace
 

